apps/dcos_dns/src/dcos_dns.erl

```diff
@@ -24,20 +19,62 @@ family(IP) when size(IP) == 8 ->
 
 -spec(get_leader_addr() -> {ok, inet:ip_address()} | {error, term()}).
 get_leader_addr() ->
-    resolve_mesos("leader.mesos").
+    case resolve(<<"leader.mesos">>, inet) of
+        {ok, [IP]} ->
+            {ok, IP};
+        {ok, _IPs} ->
+            {error, not_found};
+        {error, Error} ->
+            {error, Error}
+    end.
+
+-spec(resolve(binary()) -> {ok, [inet:ip_address()]} | {error, term()}).
+resolve(DNSName) ->
+    resolve(DNSName, inet).
+
+-spec(resolve(binary(), inet | inet6) ->
+    {ok, [inet:ip_address()]} | {error, term()}).
+resolve(DNSName, Family) ->
+    resolve(DNSName, Family, 5000).
+
+-spec(resolve(binary(), inet | inet6, timeout()) ->
+    {ok, [inet:ip_address()]} | {error, term()}).
+resolve(DNSName, Family, Timeout) ->
+    DNSNameStr = binary_to_list(DNSName),
+    ParseFun =
+        case Family of
+            inet -> fun inet:parse_ipv4_address/1;
+            inet6 -> fun inet:parse_ipv6_address/1
+        end,
+    case ParseFun(DNSNameStr) of
+        {ok, IP} ->
+            {ok, [IP]};
+        {error, einval} ->
+            imp_resolve(DNSName, Family, Timeout)
+    end.
 
--spec(resolve_mesos(inet_res:dns_name()) ->
+-spec(imp_resolve(binary(), inet | inet6, timeout()) ->
     {ok, inet:ip_address()} | {error, term()}).
-resolve_mesos(DNSName) ->
-    Opts = [{nameservers, dcos_dns_config:mesos_resolvers()}],
-    case inet_res:resolve(DNSName, in, a, Opts, 1000) of
-        {ok, DnsMsg} ->
-            case inet_dns:msg(DnsMsg, anlist) of
-                [DnsRecord] ->
-                    {ok, inet_dns:rr(DnsRecord, data)};
-                _DnsRecords ->
-                    {error, not_found}
+imp_resolve(<<"localhost">>, inet, _Timeout) ->
+    {ok, [{127, 0, 0, 1}]};
+imp_resolve(DNSName, Family, Timeout) ->
+    Type =
+        case Family of
+            inet -> ?DNS_TYPE_A;
+            inet6 -> ?DNS_TYPE_AAAA
+        end,
+    Query = #dns_query{name = DNSName, type = Type},
+    Message = #dns_message{rd = true, qc = 1, questions = [Query]},
+    Request = dns:encode_message(Message),
+    case dcos_dns_handler:resolve(udp, Request, Timeout) of
+        {ok, Response} ->
+            try dns:decode_message(Response) of #dns_message{answers = RRs} ->
```

Comment:
There should be a newline after `of`.

Reply:
IIRC, we have been using that code style to avoid linter errors because our config has very strict rule for that particular case. https://github.com/dcos/dcos-net/blob/6b41175/rebar.config#L193